pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -172,221 +168,192 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
-		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, "system")
+		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, metav1.TenantSystem)
 
 	} else {
 		// for cluster-wide resources
 		dynamicClientResource = dynamicClient.Resource(mapping.Resource)
 	}
 
 	// Create CRD resource - vpc or subnet
-	actualObject, err := dynamicClientResource.Create(unstructuredObj, metav1.CreateOptions{})
+	_, err = dynamicClientResource.Create(unstructuredObj, metav1.CreateOptions{})
 
-	if err == nil {
-		klog.V(4).Infof("Get actual object's name : (%s)", actualObject.GetName())
-		klog.V(4).Infof("Get actual object's GVK : (%v)", actualObject.GroupVersionKind())
-		klog.V(4).Infof("Get actual object's objectKind : (%v)", actualObject.GetObjectKind())
-	} else {
-		klog.Errorf("Create actual object's name: (%s) in error (%v).", unstructuredObj.GetName(), err)
+	if err != nil {
+		klog.Errorf("Create unstructed object (%s) result in error (%v).", unstructuredObj.GetName(), err)
 	}
 
 	return err
 }
 
-func GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant string) ([]byte, error) {
-	// For updating the data in default vpc/subnet template
-	var availableData = map[string]string{
-		"Tenant": tenant,
+func generateVPCSpec(vpcName string) (int, *MizarVPC) {
+	// randomize ip start segment:
+	ipStart := ran.Intn(255) + 1 	// IpStart range [1, 255]
+	// Simply not allow ip start from 10, 172, 192, 100 for well-known private range
+	if ipStart == 10 || ipStart == 172 || ipStart == 192 || ipStart == 100 {
+		ipStart++
 	}
 
-	// Read template file
-	jsonTmpl, err := readTemplateFile(defaultTemplatePath)
-	if err != nil {
-		klog.Errorf("Read default vpc/subnet template in error: (%v)", err)
-		return nil, err
+	vpc := &MizarVPC{
+		TypeMeta: TypeMeta{
+			APIVersion: "mizar.com/v1",
+			Kind: "Vpc",
+		},
+		Metadata: ObjectMeta{
+			Name: vpcName,
+		},
+		Spec:     MizarVPCSpec{
+			IP:      fmt.Sprintf("%d.0.0.0", ipStart),
+			Prefix:  "16",
+			Divider: 1,
+			Status:  "Init",
+		},
 	}
 
-	// Create Template with template file
-	t, err := template.New(vpcOrSubnetName).Parse(jsonTmpl)
-	if err != nil {
-		klog.Errorf("Create Template with template file in error: (%v)", err)
-		return nil, err
-	}
-
-	// Create json file in bytes format
-	var bytesJson bytes.Buffer
-	if err = t.Execute(&bytesJson, availableData); err != nil {
-		klog.Errorf("Update default vpc/subnet template in error: (%v)", err)
-		return nil, err
-	}
-
-	return bytesJson.Bytes(), nil
+	return ipStart, vpc
 }
 
-func readTemplateFile(path string) (string, error) {
-	bytes, err := ioutil.ReadFile(path)
-	if err != nil {
-		klog.Errorf("Read Template File (%s) in error :(%v)", path, err)
-		return "", err
+func generateSubnetSpec(vpcName, subnetName string, vpcIpStart int) ([]byte, error) {
+	subnetIpSeg := ran.Intn(256)	// 0-255
```

Comment:
256 (/24) seems a bit too limiting. Can we make it larger?

Reply:
Increased ip range for VPC and subnet.